test/changeset_upload_observation.js

```diff
@@ -0,0 +1,179 @@
+var test = require('tape')
+var contentType = require('content-type')
+var parsexml = require('xml-parser')
+var hyperquest = require('hyperquest')
+var concat = require('concat-stream')
+var isISODate = require('isostring')
+var schema = require('mapeo-schema')
+
+var createServer = require('./lib/test_server.js')
+
+var osm, base, server, changeId
+
+test('changeset_upload.js: setup server', function (t) {
+  createServer(function (d) {
+    base = d.base
+    osm = d.osm
+    server = d.server
+    t.end()
+  })
+})
+
+var obs = {
+  type: 'observation',
+  lat: 1,
+  lon: 2,
+  tags: {
+    obs: 'bar'
+  },
+  attachments: {
+    type: 'image/jpeg',
+    id: '1'
+  },
+  metadata: {
+    'hello': 'world'
+  },
+  schemaVersion: 3
+}
+
+var version, id
+
+test('create observation', function (t) {
+  t.plan(1)
+  osm.create(obs, (err, info) => {
+    t.error(err)
+    version = info.version
+    id = info.id
+    t.end()
+  })
+})
+
+test('create changeset upload', function (t) {
+  t.plan(4)
+  var href = base + 'changeset/create'
+  var hq = hyperquest.put(href, {
+    headers: { 'content-type': 'text/xml' }
+  })
+  hq.once('response', function (res) {
+    t.equal(res.statusCode, 200, 'create 200 ok')
+    var contentObj = contentType.parse(res)
+    t.equal(contentObj.type, 'text/plain', 'media type correct')
+    t.equal(contentObj.parameters.charset.toLowerCase(), 'utf-8', 'charset correct')
+  })
+  hq.pipe(concat({ encoding: 'string' }, function (body) {
```

Comment:
This is kinda off-topic, but gosh am I feeling over these super verbose hyperquest tests :P

Reply:
haha yea ...

Reply:
Agreed. Makes it hard to know what tests are testing.